File: rezmate_backend/YemenBooking.Core/Entities/Currency.cs
namespace YemenBooking.Core.Entities;

using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

/// <summary>
/// كيان العملة
/// Currency entity with string primary key (Code)
/// </summary>
[Display(Name = "كيان العملة")]
public class Currency
{
    /// <summary>
    /// رمز العملة (المعرف الأساسي)
    /// Currency code (primary key)
    /// </summary>
    [Key]
    [Display(Name = "رمز العملة")]
    public string Code { get; set; }

    /// <summary>
    /// الرمز باللغة العربية
    /// Arabic currency code
    /// </summary>
    public string ArabicCode { get; set; }

    /// <summary>
    /// اسم العملة
    /// Currency name
    /// </summary>
    public string Name { get; set; }

    /// <summary>
    /// اسم العملة بالعربية
    /// Arabic currency name
    /// </summary>
    public string ArabicName { get; set; }

    /// <summary>
    /// هل هي العملة الافتراضية
    /// Whether this is the default currency
    /// </summary>
    public bool IsDefault { get; set; }

    /// <summary>
    /// سعر الصرف بالنسبة للعملة الافتراضية
    /// Exchange rate relative to default currency; null for default
    /// </summary>
    public decimal? ExchangeRate { get; set; }

    /// <summary>
    /// تاريخ آخر تحديث لسعر الصرف
    /// Last update timestamp for the exchange rate
    /// </summary>
    public DateTime? LastUpdated { get; set; }

    /// <summary>
    /// عقارات تستخدم هذه العملة (ملاحة عكسية)
    /// Properties that use this currency (reverse navigation via Property.Currency)
    /// </summary>
    public virtual ICollection<Property> Properties { get; set; } = new List<Property>();
}

File: hggzkportal_app/lib/features/admin_currencies/domain/entities/currency.dart
import 'package:equatable/equatable.dart';

class Currency extends Equatable {
  final String code;
  final String arabicCode;
  final String name;
  final String arabicName;
  final bool isDefault;
  final double? exchangeRate;
  final DateTime? lastUpdated;

  const Currency({
    required this.code,
    required this.arabicCode,
    required this.name,
    required this.arabicName,
    required this.isDefault,
    this.exchangeRate,
    this.lastUpdated,
  });

  Currency copyWith({
    String? code,
    String? arabicCode,
    String? name,
    String? arabicName,
    bool? isDefault,
    double? exchangeRate,
    DateTime? lastUpdated,
  }) {
    return Currency(
      code: code ?? this.code,
      arabicCode: arabicCode ?? this.arabicCode,
      name: name ?? this.name,
      arabicName: arabicName ?? this.arabicName,
      isDefault: isDefault ?? this.isDefault,
      exchangeRate: exchangeRate ?? this.exchangeRate,
      lastUpdated: lastUpdated ?? this.lastUpdated,
    );
  }

  @override
  List<Object?> get props => [
        code,
        arabicCode,
        name,
        arabicName,
        isDefault,
        exchangeRate,
        lastUpdated,
      ];
}